### DISCUSSION TITLE
Snapping object in real world

### DISCUSSION THREAD

**Discussion by ManChrys** [microsoft/MixedRealityToolkit-Unity#10851] [Q&A]
Hello guys,

i think that here is the best place to ask. I would like to ask how can i create a mesure tool which will be measure only horizontally and vertically. If i will use two spheres and a line renderer as you can see on the photo below, how can i snap the two spheres on real world so i will have better accuracy ?

![1](https://user-images.githubusercontent.com/67521662/183298923-c4cfea15-1f69-44a7-9344-6fc61dd1ca18.png)

something like that but using a measuretool.

https://www.microsoft.com/en-us/research/video/snaptoreality-aligning-augmented-reality-real-world-2/

Thanks in advance.

**Comment by RogPodge** [comment]
Hey there.

We think that this component should be what are looking for to achieve this goal.
https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk2/features/ux-building-blocks/solvers/tap-to-place?view=mrtkunity-2022-05

If you want further customization, you can work on top of this, which is what the tap to place component is built on top of
https://docs.microsoft.com/en-us/windows/mixed-reality/design/surface-magnetism

**Reply by ManChrys** [reply]
Hello and thanks for the reply.

I tried the Surface magnetism but all i want is to snap the center of the sphere to an edge so i have better accuracy on measurements, something like that

https://www.youtube.com/watch?v=bCvxKdw-mPw

Thanks in advance.
